### ISSUE TITLE
NetNewsWire Web Content with high CPU utilization on Mac

### ISSUE THREAD

**Issue by kleinhirsch** [Ranchero-Software/NetNewsWire#3330]
NetNewsWire (Version 6.0.3 6035) is using all my CPU cores. Activity Monitor is showing about 50% on each core. I have only noticed this behaviour after installing Monterey (macOS 12.0.1). A simple test I did was to remove all subscriptions and turn off update checks. This did not have any impact on the CPU utilization. 

Is there any way for me to find out what's going on under the hood?

**Comment by vincode-io** [comment]
Is it only certain articles that you are viewing that cause this problem or is it all the time?

**Comment by kleinhirsch** [comment]
This is all the time, right after starting the software. And this happens even after deleting all subscriptions.So there is only a blank window visible.

**Comment by vincode-io** [comment]
Is it the NetNewsWire Web Content process that is pegging your CPU's or is it the main NetNewsWire process?

**Comment by kleinhirsch** [comment]
It's NetNewsWire Web Content. NetNewsWire sits at 0 or 0.2% and doing fine.

**Comment by vincode-io** [comment]
We don't have any code in that process. It is all Apple code. That process has gotten confused some how, so let's try to get it straightened out.

Have you rebooted your computer since this started happening? If not, that is your first step.

If that doesn't work, let's try deleting any cached files that might be causing a problem. Open the terminal and run `~/Library/Containers/com.ranchero.NetNewsWire-Evergreen` for me. Then start up NetNewsWire and see if you still see the same problem.

**Comment by kleinhirsch** [comment]
Restarting had no effect. After that I moved the `com.ranchero.NetNewsWire-Evergreen` folder to a different location, so NetNewsWire had to create a fresh one. That also had no effect.

**Comment by vincode-io** [comment]
Let's try deleting the caches next.

`rm -rf ~/Library/Caches/com.ranchero.NetNewsWire-Evergreen`

**Comment by kleinhirsch** [comment]
That had no effect, either.

**Comment by vincode-io** [comment]
Let's see if there are any clues by sampling the process. In the Activity Monitor, select NetNewsWire Web Content. Then in the toolbar, click the button with the ellipsis in a circle towards the middle, and select Sample Process.

When that is done, there will be a Save button in the top right of the window that just opened. Attach that to this ticket.

Maybe we'll get lucky and see something.

**Comment by kleinhirsch** [comment]
[NetNewsWire Web Content.txt](https://github.com/Ranchero-Software/NetNewsWire/files/7449587/NetNewsWire.Web.Content.txt)

**Comment by vincode-io** [comment]
Unfortunately, there is nothing in there that is useful. Let's try one last thing.

Make sure there are no NetNewsWire processes running. Kill them in the Activity Monitor if you have to. Let's delete the cache's again.

`rm -rf ~/Library/Caches/com.ranchero.NetNewsWire-Evergreen`

Then immediately do a restart and hope for the best.

**Comment by kleinhirsch** [comment]
Sadly this still had no effect. If I find something, I will report back.
Thank you for all your help!

**Comment by vincode-io** [comment]
Sorry that I wasn't any help. I think something must have gone wrong with your 12.0.1 update.

If it happened to me, I'd install the latest Beta to see if that helps. If that didn't, a full reinstall would be in order. 😞

**Comment by lolbat** [comment]
I m having the same issue in my M1 Macbook Air. My usage is usually at 7-8%. Often after I have cleared my Unread list. There are no articles being viewed which is the really odd thing. I just came to report this because I have a CPU monitor installed to help track down some issues with another app and I noticed the usage stats for the Web Content viewer. 

If you need anything else tried let me know

**Comment by phillco** [comment]
I'm also seeing this. "NetNewsWire Web Content" is the number one user of CPU on my M1 Mac Mini, even when its window is in the background (and not even showing an article).

![Screen Shot 2021-12-25 at 1 54 52 PM](https://user-images.githubusercontent.com/536668/147394183-6c545298-1149-49ea-b409-47c8ac1a66be.png)

![Screen Shot 2021-12-25 at 1 53 11 PM](https://user-images.githubusercontent.com/536668/147394164-6f7d6030-c3b9-4215-b5c1-47d6437b3097.png)

![Screen Shot 2021-12-25 at 1 53 50 PM](https://user-images.githubusercontent.com/536668/147394169-2b9cb712-e6f7-4fc2-9176-ef0fff284f00.png)

This seems to be 100% reproducible; it always happen when I run NNW, and doesn't seem to ever stop.

**Comment by vincode-io** [comment]
@phillco Does it happen right when you launch NNW or are you reading some content before noticing it?

**Comment by phillco** [comment]
@vincode-io it happens as soon as I launch

**Comment by vincode-io** [comment]
Which version of macOS are you on?

**Comment by phillco** [comment]
12.1, but I believe this happened on Big Sur as well.

**Comment by vincode-io** [comment]
I'm not able to reproduce this and we're not sure why this is happening for some users and not others. Can you think of anything unusual on your system that might be different than other users? Are you running an early release of Safari or something?

**Comment by phillco** [comment]
No, in fact I don't even use Safari at all. It's equally confusing to me! It's also been happening consistently for a while.

I'll have to see if I can eventually narrow it down over time, but yeah, very strange.

**Comment by kleinhirsch** [comment]
It's not only a NNW thing for me. I found a similar problem with Apple Mail. Having the Mail window open will spawn a "Mail Web Content" process with high CPU load, but in contrast to NNW the process will go away if I close the Mail window. So something on my machine is making WebViews misbehave?

**Comment by vincode-io** [comment]
@kleinhirsch I do suspect it is something different on your machine, but I can't figure out what it might be. I've done a bunch of web research and come up empty handed.

Is anyone using any Safari extensions? In theory they shouldn't be causing any problems, but I think we should at least see if they could be causing the problem.

**Comment by kleinhirsch** [comment]
Turning off all Safari extensions was something I did try some time ago with no effect. The next thing I did was to not just uncheck them, but remove them from my machine (including programs with bundled extensions!). Safaris extension list was completely blank and there was still a CPU spike for Mail and NNW.

**Comment by brentsimmons** [comment]
Are you still seeing this bug?

**Comment by kleinhirsch** [comment]
Yes - even after a fresh install and migration from backup! So I am pretty sure NNW is not at fault and is only showing a symptom of an underlying problem. My best bet is a corrupted settings file.

**Comment by brentsimmons** [comment]
This doesn’t appear to be something we can control, unfortunately.
